From a8fc83929f07981a3b0e853e3cf45178a07b59a0 Mon Sep 17 00:00:00 2001
From: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
Date: Wed, 8 Mar 2023 00:28:09 -0800
Subject: [PATCH] ath12k: Set correct tx_bw when EHT OP is present

Allow correct phymode to be set when
EHT320 is present.

Signed-off-by: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
---
 net/mac80211/vht.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/vht.c b/net/mac80211/vht.c
index 63b2181..001b533 100644
--- a/net/mac80211/vht.c
+++ b/net/mac80211/vht.c
@@ -345,7 +345,7 @@ ieee80211_sta_cap_rx_bw(struct link_sta_info *link_sta)
 		 * specific check for now.
 		 */
 		if (eht_cap->has_eht &&
-		    link_conf->chandef.chan->band != NL80211_BAND_6GHZ) {
+		    link_conf->chandef.chan->band != NL80211_BAND_2GHZ) {
 			info = eht_cap->eht_cap_elem.phy_cap_info[0];
 
 			if (info & IEEE80211_EHT_PHY_CAP0_320MHZ_IN_6GHZ) {
-- 
2.7.4

